packages/execution-environments/src/common/endowments/buffer.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * Returns a `Buffer` endowment. This mainly exists so that our build system
+ * will reliably inline a `Buffer` implementation when targeting browser
+ * environments.
+ *
+ * @returns An object with a cross-platform `Buffer` property.
+ */
+const createBuffer = () => {
+  return { Buffer } as const;
```

Comment:
Will this work cross-platform? 🤔 Doesn't it need a polyfill in the browser?

Reply:
It does, and we do so using this Webpack plugin (see `webpack.config.js`): https://www.npmjs.com/package/node-polyfill-webpack-plugin